survey_question_type_five_star/static/src/js/survey.js

```diff
@@ -1,13 +1,44 @@
 /* Copyright 2018 ACSONE SA/NV
 License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).*/
-
-$(document).ready(function () {
+odoo.define('survey_question_type_five_star.survey', function(require) {
     'use strict';
-    $(".rate > label").click(function () {
-        var label_items = $( this ).parent().find("label");
-        var value = label_items.length - $( this ).index();
-        label_items.removeClass("checked");
-        label_items.slice($( this ).index()).addClass("checked");
-        $( this ).parent().find("input").val(value);
+    $(document).ready(function() {
+        //Print Result case
+        var the_form = $('.js_surveyform');
+        var prefill_controller = the_form.attr("data-prefill");
+        if (!the_form.length) {
+            return $.Deferred().reject("DOM doesn't contain '.js_surveyform'");
+        }
+        function prefill() {
+            if (!_.isUndefined(prefill_controller)) {
+                var prefill_def = $.ajax(prefill_controller, {
```

Comment:
This call is already done in the base module. Can't we just add a onchange listener on the input? Then when the base module updates inputs value, you can determine which star should be highlighted. I think we could use the ```Widget``` class, this widget would take a DOM element/JS selector in parameter. And the target object would be the HTML structure of the "Five star rate" representation. When the widget is instanciated, it would check the current value of the input to highlight stars, and it would add an onchange listener to handle the possible changes. The widget should be readonly for the result view and editable when answering.